### ISSUE TITLE
Feature Request: On the desktop font, make the 'fa-' prefix part of the icon names

### ISSUE THREAD

**Issue by CKHarwood** [FortAwesome/Font-Awesome#17087]
**Is your feature request related to a problem? Please describe.**
There is no problem; this idea might make the desktop font ligatures easier to use for some people.

**Describe the solution you'd like**
Currently, the invocation for Font Awesome icons in the desktop font (_e.g._, `arrow-right`) differs from the invocation used elsewhere (_e.g._, `fa-arrow-right`). I would like the desktop font to use the `fa-` prefix on icon names. Some advantages to requiring the prefix:

1. Fewer errors when flipping between using Font Awesome in code and when using Font Awesome in design. There is no problem with Font Awesome itself. There is a problem with me, with my muscle memory and context switching. I use the wrong icon "name" in different contexts, and the time I lose to my mistakes outpaces the time I gain not having to type the `fa-` prefix in design work.

2. Users bothered by the "icon flickering" as the type their way through one icon name to another icon name could omit the `fa-` prefix until they are prepared to invoke the icon.

3. It is clearer in plain text whether a word represents a Font Awesome icon or is just a word in the document. This opens those documents to find-and-replace work, which isn't unusual or particularly dangerous in early document design work.

Downside: This idea is not backwards-compatible with documents using the current font, and I imagine there are not enough desktop font users to merit maintaining two separate forks. This is more an idea for FA 6 or 7.

**Describe alternatives you've considered**
* Being smarter. This alternative has eluded me so far.
* Learning more about how font designers work and attempting to port Font Awesome to a new font as an exercise. Someday, maybe.

**Additional context**
N/A

**Feature request checklist**
- [x] This is a single feature (i.e. not a re-write of all of Font Awesome)
- [x] The title starts with "Feature request: " and is followed by a clear feature name (Ex: `Feature request: moar cowbell`)
- [x] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome-Pro/issues) and to the best of my knowledge this is not a duplicate

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community and thanks for this report.

I can see the point, @robmadole @sensibleworld any thoughts?

**Comment by robmadole** [comment]
Great idea!

> Fewer errors when flipping between using Font Awesome in code and when using Font Awesome in design. There is no problem with Font Awesome itself. There is a problem with me, with my muscle memory and context switching. I use the wrong icon "name" in different contexts, and the time I lose to my mistakes outpaces the time I gain not having to type the fa- prefix in design work.

These are exactly the type of automatic things that we are targeting for FA 6. We want to reduce friction and whenever possible lean into muscle memory. I'm adding this feature request to the 6 list.

> Downside: This idea is not backwards-compatible with documents using the current font, and I imagine there are not enough desktop font users to merit maintaining two separate forks. This is more an idea for FA 6 or 7.

I actually think it _is_ backward-compatible. The feature we use to make this happen is called ligatures. You can have multiple ligatures defined. For example, we have `times` and `TIMES`. So if you have the caps key on it still works.

That being said, it will be a feature we consider for 6.

**Comment by tagliala** [comment]
> I actually think it is backward-compatible. 

Yes, sorry, I forgot about multiple ligatures for each symbol.  Removing the label
